src/main/scala/mesosphere/marathon/api/v2/TasksResource.scala

```diff
@@ -116,22 +132,23 @@ class TasksResource @Inject() (
     if (scale && wipe) throw new BadRequestException("You cannot use scale and wipe at the same time.")
 
     val taskIds = (Json.parse(body) \ "ids").as[Set[String]]
-    val tasksToAppId: Map[String, PathId] = taskIds.map { id =>
-      try { id -> Task.Id.runSpecId(id) }
-      catch { case e: MatchError => throw new BadRequestException(s"Invalid task id '$id'.") }
+    val tasksIdToAppId: Map[Instance.Id, PathId] = taskIds.map { id =>
+      try { Task.Id(id).instanceId -> Task.Id.runSpecId(id) }
+      catch { case e: MatchError => throw new BadRequestException(s"Invalid task id '$id'. [${e.getMessage}]") }
     }(collection.breakOut)
-    def scaleAppWithKill(toKill: Map[PathId, Seq[Instance]]): Response = {
-      deploymentResult(result(taskKiller.killAndScale(toKill, force)))
+
+    def scaleAppWithKill(toKill: Map[PathId, Seq[Instance]]): Future[Response] = async {
+      val killAndScale = await(taskKiller.killAndScale(toKill, force))
+      deploymentResult(killAndScale)
     }
 
-    def killTasks(toKill: Map[PathId, Seq[Instance]]): Response = {
-      val affectedApps = tasksToAppId.values.flatMap(appId => result(groupManager.app(appId))).toSeq
+    def killTasks(toKill: Map[PathId, Seq[Instance]]): Future[Response] = async {
+      val affectedApps = await(Future.sequence(tasksIdToAppId.values.map(appId => groupManager.app(appId)).toSeq)).flatten
```

Comment:
you can replace ".map().toSeq()" with ".map()(collection.breakOut)" - you may need to capture it in a val annotated with the type in order to do so.